src/napari_spatialdata/_sdata_widgets.py

```diff
@@ -1,53 +1,74 @@
 from __future__ import annotations
 
+from collections import Counter
 from typing import TYPE_CHECKING, Iterable
 
 import shapely
+from napari.utils.events import EventedList
 from qtpy.QtWidgets import QLabel, QListWidget, QListWidgetItem, QVBoxLayout, QWidget
 from spatialdata import SpatialData
 
 from napari_spatialdata._viewer import SpatialDataViewer
+from napari_spatialdata.utils._utils import _get_sdata_key
 
 if TYPE_CHECKING:
     from napari import Viewer
     from napari.utils.events.event import Event
 
 
 class ElementWidget(QListWidget):
-    def __init__(self, sdata: SpatialData):
+    def __init__(self, sdata: EventedList):
         super().__init__()
         self._sdata = sdata
 
     def _onClickChange(self, selected_coordinate_system: QListWidgetItem | int | Iterable[str]) -> None:
         self.clear()
 
         elements = {}
-        for element_type, element_name, _ in self._sdata.filter_by_coordinate_system(
-            selected_coordinate_system
-        )._gen_elements():
-            elements[element_name] = element_type
+        element_names = [element_name for sdata in self._sdata for _, element_name, _ in sdata._gen_elements()]
+        element_names = [element for element, count in Counter(element_names).items() if count > 1]
```

Comment:
please rename to duplicate_element_names

Reply:
Resolved with [d2d66f3](https://github.com/scverse/napari-spatialdata/pull/136/commits/d2d66f32507a57d1adcecd1c93022e02b49ebc4d)